src/cdk/a11y/focus-trap/focus-trap-manager.ts

```diff
@@ -0,0 +1,61 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Injectable} from '@angular/core';
+
+/**
+ * A FocusTrap managed by FocusTrapManager.
+ * Implemented by ConfigurableFocusTrap to avoid circular dependency.
+ */
+export interface ManagedFocusTrap {
+    enable(): void;
+    disable(): void;
+    focusInitialElementWhenReady(): void;
```

Comment:
This one's `Promise<boolean>` in the current focus trap. The boolean indicates whether focus was trapped and it needs to be a promise because it won't trap focus immediately.